CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/components/LayersViewPane.java

```diff
@@ -0,0 +1,286 @@
+        Platform.runLater(() -> {
+            controller.getVxQueryCollection().setDefaultQueries();
+            controller.getTxQueryCollection().setDefaultQueries();
+            options.displayQueryErrorLabel(false);
+            
+            final VBox oldLayers = new VBox();
+            createLayers(oldLayers, controller.getVxQueryCollection().getQueries(), controller.getTxQueryCollection().getQueries());
+            attributeScrollPane.setContent(oldLayers);
+            this.layersViewVBox = new VBox(layersHeading, attributeScrollPane, options);
+            // trigger refresh using enabled method
+            setEnabled(true);
+            cdl1.countDown();
+        });
+        try {
+            cdl1.await();
+        } catch (InterruptedException ex) {
+            LOGGER.log(Level.WARNING, ex.getMessage());
+            Thread.currentThread().interrupt();
+        } 
+    }
+
+    /**
+     * Set the pane enabled will switch between the real layers pane and a
+     * message pane notifying the user that a graph is required.
+     *
+     * @param enable true if there is a graph
+     */
+    public void setEnabled(final boolean enable) {
+        Platform.runLater(() -> this.setCenter(enable ? layersViewVBox : noGraphPane));
+    }
+
+    /**
+     * Recreate the layers
+     *
+     * @param layersnew
+     * @param vxQueries
+     * @param txQueries
+     */
+    private void createLayers(final VBox layersnew, final BitMaskQuery[] vxQueries, final BitMaskQuery[] txQueries) {
+        final int iteratorEnd = Math.max(getHighestQueryIndex(vxQueries), getHighestQueryIndex(txQueries)) + 1;
+        boolean isQueryActive = false;
+        
+        for (int position = 0; position < iteratorEnd; position++) {
+            final BitMaskQuery vxQuery = vxQueries[position];
+            final BitMaskQuery txQuery = txQueries[position];
+            if (vxQuery == null && txQuery == null) {
+                continue;
+                // TODO: Can we break here? both null meaning no layer - should be in order.
+            }
+
+            final int queryIndex = vxQuery != null ? vxQuery.getIndex() : txQuery.getIndex();
+            final boolean queryVisibility = vxQuery != null ? vxQuery.isVisible() : txQuery.isVisible();
+            final String vxqueryString = vxQuery != null ? vxQuery.getQueryString() : StringUtils.EMPTY;
+            final String txqueryString = txQuery != null ? txQuery.getQueryString() : StringUtils.EMPTY;
+
+            // Set query description to either the vertex of transaction description if not null
+            String queryDescription = StringUtils.EMPTY;
+            if (vxQuery != null) {
+                queryDescription = vxQuery.getDescription();
+            } else if (txQuery != null) {
+                queryDescription = txQuery.getDescription();
+            }
+            
+            final Query q;
+            if (StringUtils.isEmpty(txqueryString) && StringUtils.isNotEmpty(vxqueryString)) {
```

Comment:
are blank strings meant pass here? Wondering if it should be `isBlank` and `isNotBlank` respectively instead